### DISCUSSION TITLE
privacy compliance inspection

### DISCUSSION THREAD

**Discussion by hpqiu** [airsdk/Adobe-Runtime-Support#2161] [General]
Found the following problems in Xiaomi application market privacy compliance inspection, how to solve them?

android.telephony.TelephonyManager.listen(TelephonyManager.java:3791)
<---com.adobe.air.telephony.AndroidTelephonyManager.listen(AndroidTelephonyManager.java:69)
<---com.adobe.air.AndroidActivityWrapper.initCallStateListener(AndroidActivityWrapper.java:1220)
<---com.adobe.air.customHandler.callTimeoutFunction(Native Method)
<---com.adobe.air.customHandler.handleMessage(customHandler.java:22)
<---android.os.Handler.dispatchMessage(Handler.java:106)
<---android.os.Looper.loop(Looper.java:164)
<---android.app.ActivityThread.main(ActivityThread.java:6843)
<---java.lang.reflect.Method.invoke(Native Method)
<---com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
<---com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)

android.hardware.SensorManager.getSensorList(SensorManager.java:436)
<---android.hardware.SensorManager.getDefaultSensor(SensorManager.java:528)
<---android.view.OrientationEventListener.<init>(OrientationEventListener.java:71)
<---com.adobe.air.OrientationManager$1.<init>(OrientationManager.java:225)
<---com.adobe.air.OrientationManager.onActivityCreated(OrientationManager.java:224)
<---com.adobe.air.AndroidActivityWrapper.afterOnCreate(AndroidActivityWrapper.java:1027)
<---com.adobe.air.AndroidActivityWrapper.onCreate(AndroidActivityWrapper.java:914)
<---java.lang.reflect.Method.invoke(Native Method)
<---air.com.gongfubb.wkpy.AppEntry.InvokeMethod(AppEntry.java:404)
<---air.com.gongfubb.wkpy.AppEntry.InvokeWrapperOnCreate(AppEntry.java:387)
<---air.com.gongfubb.wkpy.AppEntry.onCreate(AppEntry.java:269)
<---android.app.Activity.performCreate(Activity.java:7070)
<---android.app.Activity.performCreate(Activity.java:7061)
<---android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
<---android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3067)
<---android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3191)
<---android.app.ActivityThread.-wrap11(Unknown Source:0)
<---android.app.ActivityThread$H.handleMessage(ActivityThread.java:1921)
<---android.os.Handler.dispatchMessage(Handler.java:106)
<---android.os.Looper.loop(Looper.java:164)
<---android.app.ActivityThread.main(ActivityThread.java:6843)
<---java.lang.reflect.Method.invoke(Native Method)
<---com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
<---com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)

android.hardware.SensorManager.registerListener(SensorManager.java:889)
<---android.hardware.SensorManager.registerListener(SensorManager.java:764)
<---android.view.OrientationEventListener.enable(OrientationEventListener.java:93)
<---com.adobe.air.OrientationManager.enableEventListener(OrientationManager.java:344)
<---com.adobe.air.OrientationManager.onActivityResumed(OrientationManager.java:333)
<---com.adobe.air.AndroidActivityWrapper.onResume(AndroidActivityWrapper.java:1262)
<---java.lang.reflect.Method.invoke(Native Method)
<---air.com.gongfubb.wkpy.AppEntry.InvokeMethod(AppEntry.java:404)
<---air.com.gongfubb.wkpy.AppEntry.onResume(AppEntry.java:471)
<---android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1355)
<---android.app.Activity.performResume(Activity.java:7178)
<---android.app.ActivityThread.performNewIntents(ActivityThread.java:3285)
<---android.app.ActivityThread.handleNewIntent(ActivityThread.java:3299)
<---android.app.ActivityThread.-wrap14(Unknown Source:0)
<---android.app.ActivityThread$H.handleMessage(ActivityThread.java:1999)
<---android.os.Handler.dispatchMessage(Handler.java:106)
<---android.os.Looper.loop(Looper.java:164)
<---android.app.ActivityThread.main(ActivityThread.java:6843)
<---java.lang.reflect.Method.invoke(Native Method)
<---com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
<---com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)

**Comment by henke37** [comment]
Those are stack traces, not clear problems.

**Reply by ajwfrost** [reply]
I'm assuming they're call stacks showing the point at which the privacy compliant inspector had found a problem...

**Comment by ajwfrost** [comment]
The second two there are the orientation listeners: you can already disable that if you want to ensure the SensorManager is not invoked, by using `ignore` as the value with your `<autoOrients>` setting: see https://github.com/airsdk/Adobe-Runtime-Support/issues/2014#issuecomment-1199206687

The first is triggered when the sound system is set up, so that we can check for incoming phone calls and pause/mute the audio. If you wanted to prevent this call, it would mean the audio continued through phone calls.. It's basically set up very much like:
https://stackoverflow.com/questions/11120922/how-to-pause-and-resume-audio-for-phone-calls
... I'm just reading that there may be an alternative way of doing this .. but that this no longer appears to work (or at least, wouldn't work if the user has no ring tone?). We can do some experimenting perhaps, to see if we can change this so we wouldn't need the TelephonyManager usage..

**Reply by hpqiu** [reply]
We want to disable listen for phone events.

**Reply by ac27437676** [reply]
hello,I am a chinese AdobeAir devloper,want make friend with you,can you add my QQ27437676.thank you!

**Comment by ajwfrost** [comment]
Hi @hpqiu  - please try using https://airsdk.harman.com/download/50.1.1.1 with a new flag in the app descriptor file, `<disableSensorAccess>` under the `<android>` section..
